### ISSUE TITLE
DAR objective

### ISSUE THREAD

**Issue by kun-dragon** [lancopku/CascadeBERT#1]
Good work! I'm very interested in your Difficulty-Aware Regularization. And I have read the paper.
        But I have some questions about the part:
        1. The input of classification task should be single. How do you add a difficulty-based margin objective(L(xi, xj)) that needs instance pair to final loss?
        2. When training models from n complete pre-trained and fine-tuned models {M1,.....,Mn}, should all models  be trained? And should the loss of each model add the difficulty-based margin objective?

**Comment by TobiasLee** [comment]
1. The DAR objective is implemented as a margin loss, i.e., we are comparing the confidence of different samples in a mini-batch to ensure that easy examples (i.e., the model can predict correctly) achieve higher confidence than harder examples. See https://github.com/lancopku/CascadeBERT/blob/main/cascade_bert.py#L45 for details.
2. Yes, every model in the cascade should be trained with the cross-entropy loss. All models except the strongest model, i.e., the largest model, require DAR calibration.  We don't need to add the DAR objective to calibrate the prediction of the largest model since we do not use its predictions for exiting decisions. In practice, we find that adding the DAR to the several smallest models, e.g., M1, M2, is suffficient, as they are capable of predicting almost 90% examples correctly.

**Comment by kun-dragon** [comment]
I got it. Thanks a lot! And if you have any beautiful work in Early Exit or Accelerating inference, Could you show me?

**Comment by TobiasLee** [comment]
Yes, we have collected a [paper list](https://github.com/TobiasLee/Awesome-Efficient-PLM) for improving the efficiency of pre-trained language models, and you can check the early exiting section for related work.

If you have any further questions, don't hesitate to contact me!

**Comment by kun-dragon** [comment]
Ok. Thanks!
